tests/test_util/aiohttp.py

```diff
@@ -62,11 +65,19 @@ def request(
         if params:
             url = url.with_query(params)
 
-        self._mocks.append(
-            AiohttpClientMockResponse(
-                method, url, status, content, cookies, exc, headers
-            )
+        new_response = AiohttpClientMockResponse(
+            method, url, status, content, cookies, exc, headers, long_poll
         )
+        if long_poll and content_not_empty:
+            for response in self._mocks:
+                if response.long_poll and response.match_request(method, url, params):
+                    response.set_long_poll_response(new_response)
+                    return
+            assert (
+                False
+            ), "First call with with long_poll has to be with empty content to set up queue."
+        else:
+            self._mocks.append(new_response)
```

Comment:
This all feels a little hacky. Wouldn't it make more sense to just return `new_response` (or a proxy object) if it's long poll and allow tests to add responses there ?

Reply:
what do you mean by a proxy object?

Reply:
I like that, usage example (maybe a short name for the set/add)

proxy=aioclient_mock.get("http://127.0.0.1:2020/&listen", long_poll=True)
proxy.set_long_poll_response(content)

Reply:
ok. done. also changed to queue_response

Reply:
i actually have another idea. maybe add the ability to use a side_effect on the response, so if side_effect is defined, it will return the result from the side effect function (or propagate the exception)?
this will enable an easy implementation of the long poll and will not require a lot of changes to the test_util/aiohttp.py. If we believe that there are other integrations that will use long_poll, we can add a standard handler for it.
I believe this way the code will be clearer and also there are other capabilities that it can create (e.g. normal GET that the value changes over time, also needed for qwikswitch...)

Reply:
let me know if that approach makes sense and i can implement it

Reply:
ok. just added the refactored one. It is simpler and more expandable

Reply:
I like that.

Reply:
thanks